src/render.tsx

```diff
@@ -25,17 +28,65 @@ export type RenderResult = ReturnType<typeof render>;
  */
 export default function render<T>(
   component: React.ReactElement<T>,
-  { wrapper: Wrapper, createNodeMock }: RenderOptions = {}
+  {
+    wrapper: Wrapper,
+    createNodeMock,
+    unstable_validateStringsRenderedInText,
+  }: RenderOptions = {}
 ) {
-  const wrap = (innerElement: React.ReactElement) =>
-    Wrapper ? <Wrapper>{innerElement}</Wrapper> : innerElement;
+  if (unstable_validateStringsRenderedInText) {
+    return renderWithStringValidation(component, {
+      wrapper: Wrapper,
+      createNodeMock,
+    });
+  }
+
+  const wrap = (element: React.ReactElement) =>
+    Wrapper ? <Wrapper>{element}</Wrapper> : element;
 
   const renderer = renderWithAct(
     wrap(component),
     createNodeMock ? { createNodeMock } : undefined
   );
+
+  return enrichRenderResult(renderer, wrap);
+}
+
+function renderWithStringValidation<T>(
+  component: React.ReactElement<T>,
+  {
+    wrapper: Wrapper,
+    createNodeMock,
+  }: Omit<RenderOptions, 'unstable_validateStringsRenderedInText'> = {}
+) {
+  const handleRender: React.ProfilerProps['onRender'] = (_, phase) => {
+    if (phase === 'update') {
+      assertStringsWithinText(screen.toJSON());
+    }
+  };
+
+  const wrap = (innerElement: React.ReactElement) => (
+    <Profiler id="renderProfiler" onRender={handleRender}>
+      {Wrapper ? <Wrapper>{innerElement}</Wrapper> : innerElement}
+    </Profiler>
+  );
```

Comment:
Slight tweak to match `render`
```suggestion
  const wrap = (element: React.ReactElement) => (
    <Profiler id="renderProfiler" onRender={handleRender}>
      {Wrapper ? <Wrapper>{element}</Wrapper> : innerElement}
    </Profiler>
  );
```